### ISSUE TITLE
Array: bug and request

### ISSUE THREAD

**Issue by nnbveh** [mgunyho/PdArray#14]
HI there! What a cool module!

Now, if i want to use it as a sample player(for drum samples mostly) i have a couple of problems: first, there's always an audible click in the end of the cycle, no matter which sample i load. And second, there's no consistent way to play different samples in their original speed. I guess first problem could be solved by 'fading out' the signal or somehting like that, and second by allowing the little ramp helper module to be able to automatically real the array size and set it as a middle position for it's duration knob...

And about the bug: it's the same as the one already reported i guess: when you load samples WITHOUT resizing, it crashes the rack itself. I'm on mac.

**Comment by mgunyho** [comment]
Hi, sorry for the belated reply. Thanks for letting me know of these issues.

I think I've come across clicking at the sample boundaries as well. How exactly are you playing the samples, using the miniramp module? I'll try to look into this.

I am able to reproduce the crashing when loading without resize, and will try to fix it in the near future.

**Comment by nnbveh** [comment]
Yes, i was using miniramp.

Cheers.

**Comment by mgunyho** [comment]
> there's no consistent way to play different samples in their original speed

This is true, and sometimes I find it a bit annoying too. However, using the array as a "regular sample player" falls a bit outside of its typical usage I think. There are many modules available for simple playback, the first examples that come to my mind are the Play and Player in the [cf pack](https://library.vcvrack.com/cf).

The main problem I have with this is that I can't think of an easy and intuitive way to communicate the array size to another module. I don't want to put more CV outputs, I think the front panel is pretty much full. Maybe it could be done using an expander, but I have no experience with those.

I think for now you might be better off with a more traditional sample player module if you don't need the recording or pitch bending capabilities of pdarray.

**Comment by mgunyho** [comment]
Thanks! I was also able to observe the clicking, depending on the sample. If the sample is well cut so that it both starts and ends with a zero, it will not click. But of course, this is always not the case.

One simple way to prevent clicking is to disconnect the REC POS and REC IN inputs and click the REC button once, which will set the firs sample to zero.

For a more proper solution, I think I will add a right-click menu option to miniramp to not reset the ramp after it has finished, but instead keep it at 10V. This then might cause clicking at the sample start, but maybe for a drum hit that starts immediately it's not so audible.

**Comment by mgunyho** [comment]
I have now added an option to create short fades at the sample boundaries, as well as an option to keep Miniramp at 10V after the ramp has ended. These should help with reducing clicks. I also wrote a small section on clicking prevention in the manual/readme. These changes will be available in v1.0.6, which should appear in your plugin manager soon. 

I will close this issue now, but if you encounter further issues or would like to add something to this discussion, feel free to do so here, or open a new issue! Thanks for the suggestions and feedback!
